### ISSUE TITLE
(Urgent) Regression: Terms and definitions section now unrecognizable

### ISSUE THREAD

**Issue by ronaldtse** [metanorma/metanorma-standoc#270]
With metanorma-iso 1.3.23

Source:
```
[[terms]]
=== Terms and definitions

[[terms-concepts]]
==== Basic concepts

[[term-date]]
===== date

_time_ (<<term-time>>) on the _calendar_ (<<term-calendar>>) _time scale_ (<<term-time-scale>>)

NOTE: Common forms of date include _calendar date_ (<<term-calendar-date>>), _ordinal date_ (<<term-ordinal-date>>) or _week date_ (<<term-week-date>>).
```

Previously:

```
<clause id="tda" obligation="normative"><title>Terms and definitions</title><clause id="terms" obligation="normative"><title>Terms and definitions</title><terms id="terms-concepts" obligation="normative"><title>Basic concepts</title><term id="term-date"><preferred>date</preferred><definition><p id="_77cc035b-8372-419d-8a41-bcfb936847b9"><em>time</em> (<xref target="term-time"/>) on the <em>calendar</em> (<xref target="term-calendar"/>) <em>time scale</em> (<xref target="term-time-scale"/>)</p></definition>
<termnote id="_2bec2fa4-7e55-475a-941d-88d4f5caa9ae">
  <p id="_1c395815-6556-46ae-95c8-db1f1acb2c17">Common forms of date include: <em>calendar date</em> (<xref target="term-calendar-date"/>), <em>ordinal date</em> (<xref target="term-ordinal-date"/>), and <em>week date</em> (<xref target="term-week-date"/>).</p>
</termnote></term>
```

Now:

```
<clause id="terms" inline-header="false" obligation="normative"><title>Terms and definitions</title><clause id="terms-concepts" inline-header="false" obligation="normative"><title>Basic concepts</title><clause id="term-date" inline-header="false" obligation="normative"><title>date</title><p id="_021c9970-e39e-4367-b6cf-1e1c55381fee"><em>time</em> (<xref target="term-time"/>) on the <em>calendar</em> (<xref target="term-calendar"/>) <em>time scale</em> (<xref target="term-time-scale"/>)</p>
<note id="_0492cf7f-b2d4-4935-8aef-95f989d55394">
<p id="_e2f08216-797e-4b7c-a4e0-96ca23201ab1">Common forms of date include <em>calendar date</em> (<xref target="term-calendar-date"/>), <em>ordinal date</em> (<xref target="term-ordinal-date"/>) or <em>week date</em> (<xref target="term-week-date"/>).</p>
</note></clause>
```

**Comment by ronaldtse** [comment]
From @Intelligent2013 :

> @ronaldtse looks like there is an issue in <example> vs <termexample> also.

**Comment by opoudjis** [comment]
The fix in https://github.com/metanorma/metanorma-iso/issues/313 is wrong.

**Comment by opoudjis** [comment]
The fix in metanorma/metanorma-iso#313 was only applicable if the child term section was empty; refined test to fall through to normal processing when it is not.

**Comment by opoudjis** [comment]
The logic is:

````
        node.role == "nonterm"  and
          return nonterm_term_def_subclause_parse(attrs, xml, node)
        st = sectiontype(node, false)
        return symbols_parse(attrs, xml, node) if @definitions
        sub = node.find_by(context: :section) { |s| s.level == node.level + 1 }
        sub.empty? || (return term_def_parse(attrs, xml, node, false))
        st == "symbols and abbreviated terms" and
          (return symbols_parse(attrs, xml, node))
        st == "terms and definitions" and
          return clause_parse(attrs, xml, node)
        term_def_subclause_parse1(attrs, xml, node)
````

Non-Term clauses disable `@term_ref` (special processing of terms)
Definitions clauses are processed as such
If there are children clauses, this is processed as a normal terms clause
If this is a childless clause: it is processed as a plain clause if titled as a terms clause (so that a clause is not misinterpreted as a term)
Else, it is a term clause
